Sources/Basic/StringConversions.swift

```diff
@@ -0,0 +1,76 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+/// Check if the given code unit needs shell escaping.
+//
+/// - Parameters:
+///     - codeUnit: The code unit to be checked.
+///
+/// - Returns: True if shell escaping is not needed.
+private func inShellWhitelist(_ codeUnit: UInt8) -> Bool {
+    switch codeUnit {
+        case UInt8(ascii: "a")...UInt8(ascii: "z"),
+             UInt8(ascii: "A")...UInt8(ascii: "Z"),
+             UInt8(ascii: "-"),
+             UInt8(ascii: "_"),
+             UInt8(ascii: "/"),
+             UInt8(ascii: ":"),
+             UInt8(ascii: "@"),
+             UInt8(ascii: "#"),
```

Comment:
Put another way, the whitelist should contain exactly those character for which the meaning is never changed whether or not those characters are quoted.  Otherwise, you could say the same about `$` etc:  quoting vs not quoting it does affect the semantics by changing how the shell interprets it.